src/component/auction-page/auction-page.component.jsx

```diff
@@ -7,49 +7,97 @@ import image from "../../asset/images/phone.jpg";
 
 export const AuctionPage = () => {
   const isLoggedIn = useSelector(({ user }) => user.isLoggedIn);
+  const userId = useSelector(({ user }) => user?.userData?.uid);
   const { id } = useParams();
 
   const [auction, setAuction] = useState({});
+  let twoDays = new Date(172800000);
+  let startDate = auction.startDate.seconds * 1000;
+
+  const CalculateTime = () => {
```

Comment:
can you refactor CalculateTime in the *util* section, as it can be abstracted from the component. You only care about the result.